cli/command/formatter/treewriter/treewriter.go

```diff
@@ -0,0 +1,52 @@
+package treewriter
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/docker/cli/cli/command/formatter/tabwriter"
+)
+
+func PrintTree(header []string, rows [][]string) {
+	// TODO(thaJeztah): using [][]string doesn't work well for this; we should
+	// create a type for this that has "optional" child records that we can
+	// recurse over to build the tree.
+	tree := make([][]string, 0, len(rows))
+	tree = append(tree, header)
+	tree = append(tree, rows...)
+
+	buf := bytes.NewBuffer(nil)
+	tw := tabwriter.NewWriter(buf, 20, 1, 3, ' ', 0)
+	for rowNum, cols := range tree {
+		if len(cols) == 0 {
+			continue
+		}
+
+		// The treePrefix is basically what makes it a "tree" when printing.
+		// We shouldn't need to have "columns" though, only know about nesting
+		// level, and otherwise have a pre-formatted, tab-terminated string
+		// for each row.
+		var treePrefix string
+		treePrefix, cols = cols[0], cols[1:]
```

Comment:
This is the part we need to generate to make it a tree; everything else should be just a pre-formatted, tab-terminated line.